File: app/lib/data/interaction/interaction_dto.dart
class InteractionRequest {
  const InteractionRequest({required this.sessionId, required this.message});

  final String sessionId;
  final String message;
}

class InteractionResponse {
  const InteractionResponse({
    required this.sessionId,
    required this.interactionId,
    required this.message,
    required this.answer,
  });

  final String sessionId;
  final String interactionId;
  final String message;
  final String answer;
}

File: services/game-service/service/interaction_dto.go
package service

type InteractionRequest struct {
	Message string `json:"message"`
}

type InteractionResponse struct {
	Message string `json:"message"`
}

type InteractionRecordResponse struct {
	InteractionID string `json:"interactionId"`
	SessionID     string `json:"sessionId"`
	RequestID     string `json:"requestId"`
	Message       string `json:"message"`
	Answer        string `json:"answer"`
	CreatedAt     string `json:"createdAt"`
}

type ListInteractionsResponse struct {
	Interactions []InteractionRecordResponse `json:"interactions"`
}
